tests/generic/test_util.py

```diff
@@ -23,6 +25,44 @@ def test_static_isinstance(self):
         )
         self.assertRaises(TypeError, static_isinstance, list(), 1)
 
+    def test_deprecate(self):
+        """ Function decorated with `deprecate` should raise a warning.  """
+        @deprecate
+        def foo(a):
+            return 2*a
+
+        @deprecate("use baz instead", version="0.2.0")
+        def bar(a):
+            return 4*a
+
+        with warnings.catch_warnings(record=True) as w:
+            self.assertEqual(foo(1), 2,
+                             "Decorated function does not return original "
```

Comment:
It took me a minute to deduce the third argument, but I like this a lot. I didn't realize unittests was set up this way (it makes sense though). I will need to start exploiting this!

Reply:
I am a big fan of the test message, it helps fixing tests later on a lot